### PR TITLE
UNDERTOW:

### PR DESCRIPTION
**Pull request by manolama**
- Allow loading the TLS cert and key from secrets. Much more secure.

### CODE REVIEW

File: implementation/server-undertow/src/main/java/net/opentsdb/tsd/TSDMain.java [hunk 1]
```diff
@@ -512,6 +545,198 @@ private static SSLContext contextFromKeyAndCerts(final Configuration config) {
...
+    try {
+      if (key_secret != null) {
+        if (key_secret instanceof RSAPrivateKey) {
+          key = (RSAPrivateKey) key_secret;
+        } else if (key_secret instanceof String || key_secret instanceof byte[]) {
+          final String raw_key;
+          if (key_secret instanceof String) {
+            raw_key = (String) key_secret;
+          } else {
+            raw_key = new String((byte[]) key_secret, Const.UTF8_CHARSET);
```
Comment (SeanPMiller) [inline implementation/server-undertow/src/main/java/net/opentsdb/tsd/TSDMain.java]: Is this known to always be UTF-8?

File: implementation/server-undertow/src/main/java/net/opentsdb/tsd/TSDMain.java [hunk 2]
```diff
@@ -512,6 +545,198 @@ private static SSLContext contextFromKeyAndCerts(final Configuration config) {
...
+          } else if (raw_key.contains("BEGIN RSA PRIVATE KEY")) {
+            key = parsePKCS1Key(trimPrivateKey(raw_key));
+          } else {
+            throw new IllegalStateException("Unrecognized key: " 
+                + config.getString(TLS_SECRET_KEY_KEY));
+          }
+        }
+        LOG.info("Successfully loaded private key from secret: "+  
+            config.getString(TLS_SECRET_KEY_KEY));
+      }
```
Comment (SeanPMiller) [inline implementation/server-undertow/src/main/java/net/opentsdb/tsd/TSDMain.java]: What about unhandled types?

File: implementation/server-undertow/src/main/java/net/opentsdb/tsd/TSDMain.java (hunk 1 above)
Comment (SeanPMiller) [inline implementation/server-undertow/src/main/java/net/opentsdb/tsd/TSDMain.java]: Is this known to always be UTF-8-encoded?

Comment (manolama) [reply implementation/server-undertow/src/main/java/net/opentsdb/tsd/TSDMain.java]: No it's an assumption for now until I can get it configured in the secrets provider.

File: implementation/server-undertow/src/main/java/net/opentsdb/tsd/TSDMain.java (hunk 2 above)
Comment (manolama) [reply implementation/server-undertow/src/main/java/net/opentsdb/tsd/TSDMain.java]: good catch, fixed.

**Comment by manolama** [comment]
Merged in 818f4769a33b755a35da9a1e19a254c332ed48bd

**Pull request closed without merging**